server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java

```diff
@@ -259,15 +259,32 @@ private List<V1EnvVar> buildCoreContainerEnvs(Task task) {
         coreContainerEnvs.put("SW_TASK_STEP", task.getStep().getName());
         coreContainerEnvs.put("DATASET_CONSUMPTION_BATCH_SIZE", String.valueOf(datasetLoadBatchSize));
         // support multi dataset uris
-        var datasetUri = swJob.getDataSets().stream()
-                .map(dataSet -> String.format(
-                        FORMATTER_URI_DATASET,
+        coreContainerEnvs.put("SW_DATASET_URI",
+                swJob.getDataSets().stream()
+                    .map(dataSet -> String.format(
+                                FORMATTER_URI_ARTIFACT,
+                                instanceUri,
+                                project.getId(),
```

Comment:
Change this project id from the dataset?

Reply:
you are right, fixed